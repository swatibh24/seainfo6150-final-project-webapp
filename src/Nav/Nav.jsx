import React from "react";
import { Link } from "react-router-dom";
import styles from "./Nav.module.css";

const Nav = () => {
    return(
        <nav>
          <Link to="/">
            <img
              className={styles.logo} src="http://shoppopbox.com/wp-content/uploads/2017/09/homechef-logo-square-1.jpg" alt="myRecipes logo"/>
          </Link>
          <Link className={styles.title} to="/">
            Home
          </Link>
           <Link className={styles.navTab} to="/recipes">
            All Recipes
           </Link>
          <Link className={styles.navTab} to="/category">
            All Categories
          </Link>
          <Link className={styles.navTab} to="/about">
            About Us
          </Link>
          <Link className={styles.navTab} to="/contact">
            Contact Us
          </Link>
          <Link className={styles.login} to="/login">
            Login
          </Link>
        </nav>
    )
};

export default Nav;
