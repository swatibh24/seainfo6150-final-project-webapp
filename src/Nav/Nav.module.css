nav {
    display: flex;
    align-items: center;
    padding-left: 20px;
    padding-right: 50px;
    border-bottom: 2px solid grey;
    background-color:#F4F4F4;
  }
  
  .logo{
    border-radius: 25%;
    width :100%
  }

  .title {
    text-decoration: none;
    font-family: Futura, Arial, sans-serif;
    color: #000;
    margin-left: 20px;
    font-size:12px
  }

  .login {
    text-decoration: none;
    margin-left: 80px;
    border-radius: 10px;
    font-family: Futura, Arial, sans-serif;
    color: #000;
  }

  .navTab {
      text-decoration: none;
      font-family: Futura, Arial, sans-serif;
      color: #000;

    }
  
  @media screen and (min-width: 400px) {
    .logo {
      width: 30px;
    }
    .title {
      font-size: 12px;
      margin-left: 20px;
    }
    .navTab {
      margin-left: 20px;
      font-size: 15px;
    }
    .login {
      padding: 1px 5px;
      font-size: 12px;
    }
  }
  
  @media screen and (min-width: 600px) {
    .logo {
      width: 40px;
    }
    .title {
      font-size: 15px;
      margin-left: 20px;
    }
    .navTab {
      margin-left: 20px;
      font-size: 15px;
    }
    .login {
      padding: 2px 10px;
      font-size: 15px;
    }
  }
  
  @media screen and (min-width: 900px) {
    .logo {
      width: 60px;
    }
    .title {
      font-size: 18px;
      margin-left: 20px;
    }
    .navTab {
      margin-left: 50px;
      font-size: 18px;
    }
    .login {
      padding: 5px 20px;
      font-size: 18px;
    }
  }
  
  @media screen and (min-width: 1200px) {
    .logo {
      width: 80px;
    }
    .title {
      font-size: 20px;
            margin-left: 20px;

    }
    .navTab {
      margin-left: 80px;
      font-size: 20px;
    }
    .login {
      padding: 5px 25px;
      font-size: 20px;
    }
  }
  