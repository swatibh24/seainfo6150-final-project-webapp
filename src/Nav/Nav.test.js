import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Nav from "./Nav.jsx";
import { BrowserRouter, Link } from "react-router-dom";

it("renders the Nav component correctly", () => {
  const { container } = render(<BrowserRouter><Nav /></BrowserRouter>);
  expect(container).toMatchSnapshot();
});