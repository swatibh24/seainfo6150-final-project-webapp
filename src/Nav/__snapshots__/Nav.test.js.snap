// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Nav component correctly 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      <img
        alt="myRecipes logo"
        class="logo"
        src="http://shoppopbox.com/wp-content/uploads/2017/09/homechef-logo-square-1.jpg"
      />
    </a>
    <a
      class="title"
      href="/"
    >
      Home
    </a>
    <a
      class="navTab"
      href="/recipes"
    >
      All Recipes
    </a>
    <a
      class="navTab"
      href="/category"
    >
      All Categories
    </a>
    <a
      class="navTab"
      href="/about"
    >
      About Us
    </a>
    <a
      class="navTab"
      href="/contact"
    >
      Contact Us
    </a>
    <a
      class="login"
      href="/login"
    >
      Login
    </a>
  </nav>
</div>
`;
